pvlib/iotools/panond.py

```diff
@@ -0,0 +1,148 @@
+"""
+Get .PAN or .OND file data into a nested dictionary.
+"""
+
+import io
+
+
+def num_type(value):
+    """
+    Determine if a value is float, int or a string
+    """
+    if '.' in value:
+        try:  # Detect float
+            value_out = float(value)
+            return value_out
+
+        except ValueError:  # Otherwise leave as string
+            value_out = value
+            return value_out
+
+    else:
+
+        try:  # Detect int
+            value_out = int(value)
+            return value_out
+
+        except ValueError:  # Otherwise leave as string
+            value_out = value
+            return value_out
+
+
+def element_type(element):
+    """
+    Determine if an element is a list then pass to num_type()
+    """
+    if ',' in element:  # Detect a list
+        values = element.split(',')
+        element_out = []
+        for val in values:  # Determine datatype of each value
+            element_out.append(num_type(val))
+
+        return element_out
+
+    else:
+        return num_type(element)
+
+
+def parse_panond(fbuf):
+    """
+    Parse a .pan or .ond text file into a nested dictionary.
+
+    Parameters
+    ----------
+    fbuf : File-like object
+        Buffer of a .pan or .ond file
+
+    Returns
+    -------
+    comp : Nested Dictionary
+        Contents of the .pan or .ond file following the indentation of the
+        file. The value of datatypes are assumed during reading. The value
+        units are the default used by PVsyst.
+
+    Raises
+    ------
+
+    Notes
+    -----
```

Comment:
I think it would be nice to see some comments here about the file format and assumptions and the parsing strategy.  Possibly also more explanatory code comments.  Ideally enough so that when the file structures evolves in the future, someone else can pick up where you left off.

Reply:
@adriesse I made updates to the comments/notes. Please let me know if you have any more feedback

Reply:
That looks great, thanks!